nautobot/extras/tests/test_plugins.py

```diff
@@ -122,14 +122,24 @@ def test_git_datasource_contents(self):
         """
         registered_datasources = registry.get("datasource_contents", {}).get("extras.gitrepository", [])
 
-        self.assertIn(
-            DatasourceContent(
-                name="text files",
-                content_identifier="dummy_plugin.textfile",
-                icon="mdi-note-text",
-                callback=refresh_git_text_files,
-            ),
-            registered_datasources,
+        plugin_datasource = DatasourceContent(
+            name="text files",
+            content_identifier="dummy_plugin.textfile",
+            icon="mdi-note-text",
+            weight=1000,
+            callback=refresh_git_text_files,
+        )
+
+        for datasource in registered_datasources:
+            if datasource.name == plugin_datasource.name:
+                self.assertEqual(datasource.content_identifier, plugin_datasource.content_identifier)
+                self.assertEqual(datasource.icon, plugin_datasource.icon)
+                self.assertEqual(datasource.weight, plugin_datasource.weight)
+                self.assertEqual(datasource.callback, plugin_datasource.callback)
+                return
+
+        self.assertEqual(
+            plugin_datasource, None, f"Datasource '{plugin_datasource.name}' could not be found in registry."
```

Comment:
As `DatasourceContent` is now a class, the original code was looking for a class instance inside of the `registry` list. But it was being re-initalised in the test so it was failing.

My thoughts were to test wether the list has a class with the correct attributes. If the class is found, it will test all the attributes then return ending the test. If no datasource is found, it asserts against `None` to display an error message. I see now i should have just raised and error.

Reply:
Thanks for the explanation. I think something like the following would be clearer:

```python
        for datasource in registered_datasources:
            if datasource.name == plugin_datasource.name:
                self.assertEqual(datasource.content_identifier, plugin_datasource.content_identifier)
                self.assertEqual(datasource.icon, plugin_datasource.icon)
                self.assertEqual(datasource.weight, plugin_datasource.weight)
                self.assertEqual(datasource.callback, plugin_datasource.callback)
                break  # found the matching datasource successfully
        else:  # not found
            self.fail(f"Datasource {plugin_datasource.name} not found in registered_datasources!")
```

Reply:
Add this as comments IMO. ❤️

Reply:
Yeah I had similar comment about this block. Thanks. :)